### ISSUE TITLE
FileDiffSuite should be published along with lms

### ISSUE THREAD

**Issue by manojo** [TiarkRompf/virtualization-lms-core#103]
Given that it is the standard way of testing generated code, it makes sense that FileDiffSuite is available to everyone who uses LMS. As in general it is bad practice to publish test folders and files, I suggest having a test-utils package to which FileDiffSuite belongs.

**Comment by TiarkRompf** [comment]
Good idea. It would also be good to (1) update this to something that doesn't produce deprecation warnings and (2) harmonize it with the tutorial repo.

But unfortunately we can't just put FileDiffSuite into the main (i.e. non-test) source tree because that would introduce a dependency on scala-test from the main sources, which also seems like bad practice.

**Comment by manojo** [comment]
Good point. At the moment, even the current lms build should be changed for not depending on scalatest though. Looking [here](https://oss.sonatype.org/content/repositories/snapshots/org/scala-lang/lms/lms-core_2.11/1.0.0-SNAPSHOT/lms-core_2.11-1.0.0-SNAPSHOT.pom) we see the dependency. We should indeed add an extra `% "test"` for the right scope:

```
libraryDependencies += ("org.scalatest" % "scalatest_2.11" % "2.2.2" % "test")
```

A possibility would be to create a separate project for `testutils`, which the main sources would depend on. Something along the [following lines](https://github.com/manojo/lms-utils/blob/master/build.sbt).

Removing deprecation warnings seems not too hard. It's about extending [Spec instead of Suite](https://github.com/manojo/lms-utils/blob/master/testutil/src/main/scala/lms/testutil/FileDiffSpec.scala) and [adding specific whitespaces](https://github.com/manojo/lms-utils/blob/master/util/src/test/scala/lms/util/OptionCPSSuite.scala#L145) in methods we want to be tested, as per the [documentation](http://doc.scalatest.org/2.2.4/#org.scalatest.Spec).

**Comment by manojo** [comment]
Ideally, of course, we'd much rather use [partest](https://github.com/scala/scala-partest-interface/), but it currently does not support virtualized-2.11 IIRC.

**Comment by TiarkRompf** [comment]
A `testutils` package seems like a good idea once we have things like e.g. random testing (which @GeorgOfenbeck is working on) but just for `FileDiffSuite`  it seems a bit heavyweight.

My suggestion would be to try and see if the relevant bits of `FileDiffSuite` can be factored out into a utility class that does not depend on ScalaTest. I think the tutorial repo _almost_ does this. 

Good point about the `% test`. Send a PR?
